File: Aeds 2/TP1/Palindromo.java

public class Palindromo {

	public static boolean isPalindromo(String text){
		boolean resp=true;
		String textr="";
		for(int i=text.length()-1;i>=0;i--) {
			textr+=text.charAt(i);
		}
		if(textr.equals(text)==true){
			return resp;
		}else{
			return false;
		}
	}
	public static void main(String[] args) {
		String text;
		text=MyIO.readString();
		while(text.equals("FIM")!=true){
			boolean resp;
			resp=isPalindromo(text);
			if(resp){
				MyIO.println("SIM");
			}else{
				MyIO.println("NAO");
			}
			text=MyIO.readString();
		}
		
	}
}

File: Aeds 2/TP1/Palindromo.c
#include <stdio.h>
#include <string.h>

int IsPalindromo(char str[1000]){
  int tam;
  char strr[1000];
  tam=strlen(str);
  for(int i=0;i<tam;i++){
    strr[tam-1-i]=str[i];
  }
  if(strcmp(str,strr)==0){
    return 0;
  }else{
    return 1;
  }
}

int main(void) {
  char str[1000];

  scanf(" %[^\n]",str);

  while((strcmp(str,"FIM")!=0)){
    if(IsPalindromo(str)==0){
      printf("SIM\n");
    }else{
      printf("NAO\n");
    }
    scanf(" %[^\n]",str);
    
  }
  
  return 0;
}